slackbot.go

```
@@ -0,0 +1,203 @@
+	"github.com/shomali11/slacker"
+)
+
+// Start will run the main code for the Slack->PD Bot
+func Start(config *config) error {
+	bot := slacker.NewClient(config.slack.token)
+	// defining which function handles the bot Init phase
+	bot.Init(
+		func() {
+			Init(config)
+		})
+
+	// error raised by the Bot will be handled by this function
+	bot.Err(
+		func(err string) {
+			Err(err)
+		})
+
+	// function tied to sentences sent to the Bot and starting with "open emergency" followed by some text
+	emergencyCmdDefinition := &slacker.CommandDefinition{
+		Description: "Open an EMERGENCY incident to Customer Support",
+		Handler: func(request slacker.Request, response slacker.ResponseWriter) {
+			Emergency(request, response, config)
+		},
+	}
+	bot.Command("open emergency <msg>", emergencyCmdDefinition)
+
+	// when no other "Command" matches and text is sent to the Bot, this function will be run instead
+	bot.DefaultCommand(
+		func(request slacker.Request, response slacker.ResponseWriter) {
+			Default(request, response, config)
+		})
+
+	// function run for all events received by the bot (including time ticks)
+	bot.DefaultEvent(
+		func(event interface{}) {
+			//log.Println(event)
+		})
+
+	// set the "help" message handling function
+	helpCmdDefinition := &slacker.CommandDefinition{
+		Description: "Help function",
+		Handler: func(request slacker.Request, response slacker.ResponseWriter) {
+			help(response, config)
+		},
+	}
+	bot.Help(helpCmdDefinition)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	err := bot.Listen(ctx)
+	if err != nil {
+		Err(err.Error())
+	}
+
+	return nil
+}
+
+// Init is called upon Bot creation (first startup)
+func Init(config *config) {
+	fmt.Printf("Connected!\n")
+	client := pagerduty.NewClient(config.pagerDuty.aPIKey)
+	var opts pagerduty.GetScheduleOptions
+
+	schedule, err := client.GetSchedule(config.pagerDuty.schedule, opts)
+	if err != nil {
+		trace.Wrap(err)
```

Did you want to return an error? Since this is just creating and immediately disposing of an error object w/o returning.

---

should be fixed now.